src/deepstream_pipeline_background_removal_cpu.py

```
@@ -0,0 +1,200 @@
+    exg_norm = cv2.normalize(exg, None, 0, 255, cv2.NORM_MINMAX).astype(np.uint8)
+    _thr, mask = cv2.threshold(exg_norm, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
+    kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (MORPH_KERNEL, MORPH_KERNEL))
+    mask = cv2.morphologyEx(mask, cv2.MORPH_CLOSE, kernel)
+    out = np.zeros_like(frame)
+    out[mask > 0] = frame[mask > 0]
+    return out
+
+
+def _map_bgr_from_buffer(pad: Gst.Pad, buf: Gst.Buffer) -> np.ndarray | None:
+    """Return BGR HxWx3 numpy array from a CPU buffer. Handles stride via GstVideoMeta.
+    Assumes upstream caps: video/x-raw, format=BGR
+    """
+    ok, map_info = buf.map(Gst.MapFlags.READ)
+    if not ok:
+        return None
+    try:
+        meta = GstVideo.buffer_get_video_meta(buf)
+        if meta is not None:
+            width, height = int(meta.width), int(meta.height)
+            stride = int(meta.stride[0])
+        else:
+            caps = pad.get_current_caps().get_structure(0)
+            width = int(caps.get_value("width"))
+            height = int(caps.get_value("height"))
+            stride = width * 3
+        arr = np.frombuffer(map_info.data, dtype=np.uint8)
+        expected = height * stride
+        if arr.size >= expected:
+            frame2d = arr[:expected].reshape((height, stride))
+            if stride == width * 3:
+                frame = frame2d.reshape((height, width, 3))
+            else:
+                frame = frame2d[:, : width * 3].reshape((height, width, 3)).copy()
+        else:
+            frame = arr.reshape((height, width, 3))
+        return frame
+    finally:
+        buf.unmap(map_info)
+
+
+class _Ctx:
+    def __init__(self, out_dir: str, save_every: int):
+        self.out_dir = out_dir
+        self.save_every = max(1, save_every)
+        self.counter = 0
+
+
+def background_removal_probe(pad, info, u_data: _Ctx):
+    buf = info.get_buffer()
+    if not buf:
+        return Gst.PadProbeReturn.OK
```

should we return `Gst.PadProbeReturn.OK`? In the case if buffer isn't available, smth is wrong with a pipeline. Having this will lead to the silent error